### ISSUE TITLE
[member-ordering] Order of getters and setters (accessors)

### ISSUE THREAD

**Issue by glen-84** [typescript-eslint/typescript-eslint#929]
**Repro**

```JSON
{
  "rules": {
        "@typescript-eslint/member-ordering": [
            "error",
            {
                "default": [
                    "public-static-field",
                    "protected-static-field",
                    "private-static-field",
                    "public-instance-field",
                    "protected-instance-field",
                    "private-instance-field",
                    "constructor",
                    "public-instance-method",
                    "protected-instance-method",
                    "private-instance-method",
                    "public-static-method",
                    "protected-static-method",
                    "private-static-method"
                ]
            }
        ]
  }
}
```

```TS
class Example {
    public test1;
    
    private test2 = "";

    public get test3() {
        return "example";
    }

    public constructor() {
        console.log(this.test2);
    }
}
```

**Expected Result**

Error, constructor should be declared before getters/accessors (depending on rule configuration).

**Actual Result**

No error.

**Additional Info**

TSLint issue: https://github.com/palantir/tslint/issues/3965 (ignore alphabetization aspect)
TSLint PR: https://github.com/palantir/tslint/pull/3984

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.13.0` |
| `@typescript-eslint/parser`        | `1.13.0` |
| `TypeScript`                       | `3.4.5` |
| `ESLint`                           | `6.1.0` |
| `node`                             | `10.16.0` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
maybe it'd be better to add a separate set of configs for getter/setters?

**Comment by glen-84** [comment]
Do you mean like `public/protected/private-getter` and `public/protected/private-setter` instead of just `public/protected/private-accessor`?

If so, it _would_ provide more control, at the expense of double the number of "types".

It _would_ be nice to enforce getters before setters (or vice versa), so that it's consistent, instead of ending up with `getter, setter, setter, getter`, or some other combination.

**Comment by bradzacher** [comment]
Exactly correct.

I think this rule is already beyond help in the configuration complexity department; so a few extra settings shouldn't hurt - esp if it gives a lot more control!

**Comment by G-Rath** [comment]
Personally I think that maybe just `accessor` would be fine, since

> It would be nice to enforce getters before setters (or vice versa), so that it's consistent, instead of ending up with getter, setter, setter, getter, or some other combination.

can be enforced by [`grouped-accessor-pairs`](https://eslint.org/docs/rules/grouped-accessor-pairs) , but ultimately I'm not too fussed so long as I can say "all accessors (static or otherwise) must be after the constructor but before instance methods, and all static accessors must be above instance accessors"

("accessors" here meaning just "getter & setter", not that it has to be only one entry in the array)

**Comment by Shinigami92** [comment]
Any news or progress for this feature?

**Comment by Pi-Bouf** [comment]
Yes it can be REALLY usefull :)

**Comment by ilyub** [comment]
My vote for this feature.

It is big drawback that member-ordering rule completely ignores accessors (getters, setters).

For this moment, I am using sort-class-members plugin (https://www.npmjs.com/package/eslint-plugin-sort-class-members) for accessors in combination with member-ordering for everything else. However, this plugin ignores visibility (public, protected, private).

So, I have to combine one rule that ignores accessors and another rule that ignores visibility.

It would be really good to have one rule for all.

**Comment by bradzacher** [comment]
We are a community run project. The **volunteer** maintainers spend most of their time triaging issues and reviewing PRs. This means that most issues *will not progress* unless a member of the community steps up and **champions** it.

If this issue is important to you - consider being that champion.
If not - please use the subscribe function and wait patiently for someone else to implement.

**Comment by ilyub** [comment]
Timewise, I will unfortunately have to join those waiting for a champion :(

Hopefully, the champion will have more interest in this fix if he sees interest from regular users.

**Comment by gultyaev** [comment]
I'll try my skills on making this feature happen

**Comment by bradzacher** [comment]
This will be released in v5
